libcudacxx/include/cuda/std/detail/libcxx/include/__iterator/indirectly_comparable.h

```diff
@@ -0,0 +1,51 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _LIBCUDACXX___ITERATOR_INDIRECTLY_COMPARABLE_H
+#define _LIBCUDACXX___ITERATOR_INDIRECTLY_COMPARABLE_H
+
+#ifndef __cuda_std__
+#include <__config>
+#endif // __cuda_std__
+
+#include "../__iterator/concepts.h"
+#include "../__iterator/projected.h"
+#include "../__functional/identity.h"
+
+#if defined(_LIBCUDACXX_USE_PRAGMA_GCC_SYSTEM_HEADER)
+#pragma GCC system_header
+#endif
+
+_LIBCUDACXX_BEGIN_NAMESPACE_STD
+
+#if _LIBCUDACXX_STD_VER > 17
+
+template <class _I1, class _I2, class _Rp, class _P1 = identity, class _P2 = identity>
```

Comment:
suggestion: I guess `_Rp` stands for Relational Predicate, which is a pedicate taking two or more arguments. In this case, the predicate can only take two arguments so we might consider `BinaryPredicate` or `Comp` to make the concept a bit more readable.

Reply:
Thanks a lot I am loving it